code/modules/power/generators/turbine.dm

```diff
@@ -143,21 +158,49 @@
 	if(default_deconstruction_screwdriver(user, initial(icon_state), initial(icon_state), I))
 		return TRUE
 
+/obj/machinery/power/compressor/multitool_act(mob/living/user, obj/item/I)
+	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+		return
+	if(!I.multitool_check_buffer(user))
+		return
+	var/obj/item/multitool/M = I
+	if(panel_open)
+		M.set_multitool_buffer(user, src)
+
+
 /obj/machinery/power/compressor/CanAtmosPass(turf/T)
 	return !density
 
+/obj/machinery/power/compressor/proc/trigger_overheat()
+	starter = FALSE
+	last_overheat = OVERHEAT_TIME
+	overheat -= 50
+	radio.autosay(OVERHEAT_MESSAGE, name, "Engineering", list(z))
+	playsound(src, 'sound/machines/buzz-two.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
+
```

Comment:
I recommend adding a proc to see how much time is left until its done. Something like this should work. It will return 0 if its past the overheat stage. If not, it should return how much time is left until its done overheating.
```suggestion
/obj/machinery/power/compressor/proc/time_until_overheat_done()
    return max(last_overheat + OVERHEAT_TIME - world.time, 0)
```